src/renderer/entities/metadata/lib/metadataService.ts

```diff
@@ -0,0 +1,62 @@
+import { ApiPromise } from '@polkadot/api';
+import { UnsubscribePromise } from '@polkadot/api/types';
+
+import { IMetadataService } from './common/types';
+import { Metadata, metadataModel } from '..';
+import { ChainId } from '@renderer/domain/shared-kernel';
+import storage from '@renderer/shared/api/storage';
+
+export const useMetadata = (): IMetadataService => {
+  const metadataStorage = storage.connectTo('metadata');
+
+  if (!metadataStorage) {
+    throw new Error('=== 🔴 Metadata storage in not defined 🔴 ===');
+  }
+
+  const { getAllMetadata } = metadataStorage;
+
+  const getMetadata = async (chainId: ChainId): Promise<Metadata | undefined> => {
+    const metadata = await getAllMetadata({ chainId });
+
+    if (metadata.length) {
+      const lastMetadata = metadata.reduce<Metadata>((acc, md) => {
+        if (md.version >= (acc.version || -1)) {
+          return md;
+        }
+
+        return acc;
+      }, {} as Metadata);
+
+      return lastMetadata;
+    }
```

Comment:
What are the benenefits of retaining all previously fetched versions in metadata in the cache? As I can see, we only allow to fetch current version of metadata from the cache, meaning all outdated version will effectively become useless. I suggest deleting all previous versions once we synced new one

Reply:
This solution will helps us to decode callData for transactions created with old metadata.

Reply:
So you are planning to use `IMetadataService` directly when decoding, right? Because requests to metadata made via ApiPromise wont use cache due to `params.length == 0` check

Reply:
Right, metadata service will be used directly for it. But we have to research how to detect correct version